projects/views.py

```diff
@@ -152,3 +163,51 @@ def partial_update(self, request, pk):
     #         'status': status.HTTP_200_OK
     #     }
     #     return Response(data, status=status.HTTP_204_NO_CONTENT)
+
+    @action(detail=False, methods=['get'], url_path='my')
+    def get_my_projects(self, request):
+        user = self.request.user
+        investors_projects = Project.objects.filter(investors__id=user.id, is_active=True)
+        serializer = ProjectViewSerializer(investors_projects, many=True, context={'request': request})
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    @action(detail=True, methods=['post'], url_path='invest')
+    def invest_to_project(self, request, pk=None):
+        """
+        This function allows adding an investor to a project.
+        The user must be authenticated and have an active investor linked to their account.
+        The investor is added to the project only if the project is active and does not already contain this investor
+
+        :param self: Allow an instance of a class to access its attributes and methods
+        :param request: Get the user from the request
+        :param pk: The primary key of the project.
+        :return: Response: A response with a detailed message about the result of the operation.
+        Examples:
+        Use this method by sending a POST request to the URL: /api/projects/12/invest
+        where 12 is the ID of the project you want to add an investor to.
+        """
+
+        try:
+            user = request.user
+            project = Project.objects.get(id=pk, is_active=True)
+            investor = Investor.objects.get(user=user, is_active=True)
```

Comment:
maybe it is worth using get_object_or_404

Reply:
Added ✅